### PR TITLE
UI rework: Title bar

### PR DESCRIPTION
**Pull request by mikerooni**
## What
Adds a title bar to machine UIs.

## Outcome
- A title bar has been added
- page switching for multiblocks now uses a dedicated page

## Additional Information

The UI in action:  
https://cdn.discordapp.com/attachments/1089296351906504835/1206315409905811597/new-ui-title-bar.mp4

### CODE REVIEW

File: src/generated/resources/assets/gtceu/lang/en_ud.json [hunk 1]
```diff
@@ -5033,6 +5048,7 @@
   "tagprefix.tiny_dust": "ʇsnᗡ %s ɟo ǝןıԀ ʎuı⟘",
   "tagprefix.tuff": "ǝɹO %s ɟɟn⟘",
   "tagprefix.turbine_blade": "ǝpɐןᗺ ǝuıqɹn⟘ %s",
+  "tagprefix.veridium": "ɯnıpıɹǝΛ %s",
```
Comment (screret) [inline src/generated/resources/assets/gtceu/lang/en_ud.json]: REMOVE YOUR KJS SCRIPTS

**Review by screret — CHANGES_REQUESTED** [review]

File: src/generated/resources/assets/gtceu/lang/en_ud.json (hunk 1 above)
Comment (mikerooni) [reply src/generated/resources/assets/gtceu/lang/en_ud.json]: Ahhh so this is what causes this

File: src/main/java/com/gregtechceu/gtceu/api/gui/fancy/FancyMachineUIWidget.java [hunk 2]
```diff
@@ -12,87 +13,231 @@
...
+    /*
+     * Current Page:      The page visible in the UI
+     * Current Home Page: The currently selected multiblock part's home page.
+     */
+    protected IFancyUIProvider currentPage;
+    protected IFancyUIProvider currentHomePage;
+
+    protected List<IFancyUIProvider> allPages;
+
+    protected Stack<NavigationEntry> previousPages = new Stack<>();
```
Comment (screret) [inline src/main/java/com/gregtechceu/gtceu/api/gui/fancy/FancyMachineUIWidget.java]: Isn't Deque better for most stuff?

Comment (mikerooni) [reply src/main/java/com/gregtechceu/gtceu/api/gui/fancy/FancyMachineUIWidget.java]: Hmm probably... hold on a sec

**Review by screret — APPROVED** [review]

**Review by screret — APPROVED** [review]

**Review by screret — APPROVED** [review]

**Comment by htmlcsjs** [comment]
> This content is no longer available.
:waagone:

**Pull request merged**